app/src/main/java/com/duckduckgo/app/privatesearch/PrivateSearchViewModel.kt

```diff
@@ -63,7 +73,20 @@ class PrivateSearchViewModel @Inject constructor(
     fun onAutocompleteSettingChanged(enabled: Boolean) {
         Timber.i("User changed autocomplete setting, is now enabled: $enabled")
         settingsDataStore.autoCompleteSuggestionsEnabled = enabled
-        viewModelScope.launch { viewState.emit(currentViewState().copy(autoCompleteSuggestionsEnabled = enabled)) }
+        viewModelScope.launch {
+            viewState.emit(
+                currentViewState().copy(
+                    autoCompleteSuggestionsEnabled = enabled,
+                    autoCompleteRecentlyVisitedSitesSuggestionsUserEnabled = enabled && history.isHistoryUserEnabled(),
```

Comment:
Not sure if I understand what you mean, I still need to check it to set the initial state of the toggle

Reply:
I did it this way so when autocomplete feature is off, visited sites is not only disabled, but off as well. If we want everything to happen internally, I'd need to expose some additional methods on the auto-complete API that combine both autocomplete and history and query them instead of history. However, I don't think it's worth the effort

Reply:
👍 
(note from conversation: if autocomplete disabled, we should also update history internal setting, and not only UI)